sqlite3/lib/src/wasm/file_system_v2.dart

```diff
@@ -0,0 +1,298 @@
+part of 'file_system.dart';
+
+/// A file system storing whole files in an IndexedDB database.
+///
+/// As sqlite3's file system is synchronous and IndexedDB isn't, no guarantees
+/// on durability can be made. Instead, file changes are written at some point
+/// after the database is changed. However you can wait for changes manually
+/// with [flush]
+///
+/// In the future, we may want to store individual blocks instead.
+
+class IndexedDbFileSystemV2 implements FileSystem {
```

Comment:
I like the new APIs, but is there no way of getting them into the existing indexed db implementation?

(It's ok to have mild breaking changes, the wasm API is marked as experimental and there's a documentation comment explaining that `package:sqlite3/wasm.dart` is not currently subject to semantic versioning).

Reply:
It can be simply replaced with current implementation if a breaking change is not a problem. Deprecation messages can be omitted also.